File: ts/project/.sdk/src/cmp/swift/fragment/SdkError.fragment.swift
// ProjectNameError - the SDK error type. Carries the pipeline error code, the
// originating context and cleaned result/spec snapshots.

import Foundation

public final class ProjectNameError: Error, CustomStringConvertible {
  public let isProjectNameError = true
  public let sdk = "ProjectName"
  public let code: String
  public let message: String
  public var ctx: Context?
  public var resultVal: Value = .noval
  public var specVal: Value = .noval

  public init(_ code: String, _ message: String, _ ctx: Context?) {
    self.code = code
    self.message = message
    self.ctx = ctx
  }

  public var description: String { message }
}

File: ts/project/.sdk/src/cmp/csharp/fragment/SdkError.fragment.cs
// ProjectNameError - the SDK error type. Carries the pipeline error code,
// the originating context and cleaned result/spec snapshots.

namespace ProjectNameSdk;

public class ProjectNameError : Exception
{
    public bool IsProjectNameError = true;
    public string Sdk = "ProjectName";
    public string Code;
    public Context? Ctx;
    public object? ResultVal;
    public object? SpecVal;

    public ProjectNameError(string code, string msg, Context? ctx)
        : base(msg)
    {
        Code = code;
        Ctx = ctx;
    }
}
